### PR TITLE
#426 case insensitive filtering

### PR DESCRIPTION
**Pull request by Rishi-0007**
This PR addresses **issue #426**, introducing the ability to perform case-insensitive filtering in Taipy tables. The enhancement allows users to apply case-insensitive string matching when filtering data in tables, specifically for the "contains" action, which is now updated to handle case-insensitivity by default.

#### Changes Implemented:
- Modified the filtering logic in `pandas_data_accessor.py` to make string matching case-insensitive when using the "contains" filter action.
- Updated the `TableFilter.tsx` and related components to support the new case-insensitive behavior.
- Refactored the test suite in `TableFilter.spec.tsx` to include tests for case-insensitive filtering.

#### Key Benefits:
- Provides users with more flexibility in filtering data without worrying about case sensitivity.
- Enhances the usability of Taipy tables by ensuring that common filtering operations align with user expectations.

#### Testing:
- All existing test cases have been passed, including newly added tests for case-insensitive filtering.
- Verified through unit tests in `TableFilter.spec.tsx` that string filtering behaves correctly for both lowercase and uppercase inputs.

#### Notes:
- Ensure that all changes follow the contribution guidelines and coding standards.

Please review this PR and provide feedback. Thanks!

### CODE REVIEW

**Comment by Rishi-0007** [comment]
Hi, maintainers! 

This PR implements case-insensitive filtering for string columns in Taipy tables. I followed the current structure for handling filters without introducing any breaking changes. However, while working on this, I considered adding a flag to the props to allow developers to control case sensitivity explicitly, rather than making all string filtering case-insensitive by default.

I’d love your feedback on this approach. Should we proceed with adding such a flag to make this behavior configurable, or is the current solution preferred for simplicity? Your guidance on this would be greatly appreciated!

Thanks for your time and support!

**Comment by Rishi-0007** [comment]
@quest-bot loot #426

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
As discussed in the issue, the only change in the frontend would be to add a case sensitive icon to the UI for filters on string columns and modify FilterDesc interface to carry that information to the backend.
In the backend we would want to support more than the `contains` action for strings.
Good start

**Comment by Rishi-0007** [comment]
Hello @FredLL-Avaiga ,

As per the recent discussion, I've implemented the requested changes to the `TableFilter` component:

1. **Case-insensitive filtering**: Added a toggle for users to enable/disable case-sensitive string filtering.
2. **UI enhancements**: Integrated a case sensitivity toggle in the UI, ensuring it's intuitive and accessible.
3. **Tests**: Updated and created tests to validate case-insensitive filtering functionality to ensure robustness.

 Please review the changes, and let me know if any additional adjustments are needed.

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 1]
```diff
@@ -104,15 +112,20 @@ const getFilterDesc = (columns: Record<string, ColumnDesc>, colId?: string, act?
...
                 value:
                     typeof val === "string"
-                        ? colType === "number"
-                            ? parseFloat(val)
-                            : colType === "boolean"
-                            ? val === "1"
-                            : colType === "date"
-                            ? getDateTime(val)
+                        ? colType === "string" && act === "contains"
+                            ? matchCase // Apply case-sensitive filtering based on the toggle
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for this

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 2]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
     const [colId, setColId] = useState<string>("");
     const [action, setAction] = useState<string>("");
     const [val, setVal] = useState<string>("");
+    const [matchCase, setMatchCase] = useState<boolean>(true); // Case-sensitivity state
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: I would set it to false by default.
What do you think @FabienLelaquais ?

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 3]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
     const [enableCheck, setEnableCheck] = useState(false);
     const [enableDel, setEnableDel] = useState(false);
 
+    // Function to handle case-sensitivity toggle
+    const toggleMatchCase = () => setMatchCase(!matchCase);
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: useCallback

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 4]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
         },
-        [columns, colId, val]
+        [columns, colId, val, matchCase]
     );
+
     const onValueChange = useCallback(
         (e: ChangeEvent<HTMLInputElement>) => {
-            setVal(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, colId, action, e.target.value));
+            const value = matchCase ? e.target.value : e.target.value.toLowerCase();
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for that the comparison is done on the backend

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 5]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
+            setVal(value);
+            setEnableCheck(!!getFilterDesc(columns, colId, action, value, matchCase));
         },
-        [columns, colId, action]
+        [columns, colId, action, matchCase]
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for the new dependency

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 6]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
+        [columns, colId, val, matchCase]
     );
+
     const onValueChange = useCallback(
         (e: ChangeEvent<HTMLInputElement>) => {
-            setVal(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, colId, action, e.target.value));
+            const value = matchCase ? e.target.value : e.target.value.toLowerCase();
+            setVal(value);
+            setEnableCheck(!!getFilterDesc(columns, colId, action, value, matchCase));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: pass false for matchCase

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 7]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
     );
+
     const onActSelect = useCallback(
         (e: SelectChangeEvent<string>) => {
             setAction(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, colId, e.target.value, val));
+            setEnableCheck(!!getFilterDesc(columns, colId, e.target.value, val, matchCase));
         },
-        [columns, colId, val]
+        [columns, colId, val, matchCase]
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for the new dependency

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 8]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
+    const toggleMatchCase = () => setMatchCase(!matchCase);
+
     const onColSelect = useCallback(
         (e: SelectChangeEvent<string>) => {
             setColId(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, e.target.value, action, val));
+            setEnableCheck(!!getFilterDesc(columns, e.target.value, action, val, matchCase));
         },
-        [columns, action, val]
+        [columns, action, val, matchCase]
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for the new dependency

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 9]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
-        [columns, colId, action]
+        [columns, colId, action, matchCase]
     );
+
     const onValueAutoComp = useCallback(
         (e: SyntheticEvent, value: string | null) => {
-            setVal(value || "");
-            setEnableCheck(!!getFilterDesc(columns, colId, action, value || ""));
+            const inputValue = value || "";
+            const processedValue = matchCase ? inputValue : inputValue.toLowerCase();
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: no need for this

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 10]
```diff
@@ -126,44 +139,56 @@ const FilterRow = (props: FilterRowProps) => {
...
+            setVal(processedValue);
+            setEnableCheck(!!getFilterDesc(columns, colId, action, processedValue, matchCase));
         },
-        [columns, colId, action]
+        [columns, colId, action, matchCase]
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: remove matchCase

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 11]
```diff
@@ -174,10 +199,11 @@ const FilterRow = (props: FilterRowProps) => {
     );
 
     const onDeleteClick = useCallback(() => setFilter(idx, undefined as unknown as FilterDesc, true), [idx, setFilter]);
+
     const onCheckClick = useCallback(() => {
-        const fd = getFilterDesc(columns, colId, action, val);
+        const fd = getFilterDesc(columns, colId, action, val, matchCase);
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx:200]: 👍

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 12]
```diff
@@ -283,6 +310,17 @@ const FilterRow = (props: FilterRowProps) => {
                     />
                 )}
             </Grid>
+
+            {/* Case Sensitivity Toggle */}
+            <Grid size={3}>
+                <FormControlLabel
+                    control={
+                        <Switch checked={matchCase} onChange={toggleMatchCase} data-testid="CaseSensitiveToggle" />
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: don't like the data-testid
give it a classname

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: We'd like to use the icon described in the issue

File: taipy/gui/data/pandas_data_accessor.py [hunk 13]
```diff
@@ -267,12 +267,22 @@ def __get_data(  # noqa: C901
                 col = fd.get("col")
                 val = fd.get("value")
                 action = fd.get("action")
+                match_case = fd.get("matchCase", True)  # Retrieve matchCase from the frontend payload
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: default value False

File: taipy/gui/data/pandas_data_accessor.py [hunk 14]
```diff
@@ -267,12 +267,22 @@ def __get_data(  # noqa: C901
...
+                
                 if isinstance(val, str):
-                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
-                        val = datetime.fromisoformat(val[:-1])
+                    # Handle case-sensitive/insensitive filtering for string 'contains'
+                    if action == "contains":
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py:283]: we want this to work for **any** action

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
- simplify the front end: add an toggle icon and don't touch the value
- backend: support all actions
- needs backend test

**Comment by Rishi-0007** [comment]
Hi @FredLL-Avaiga @FabienLelaquais,
I have implemented the required frontend changes as suggested. On the backend, I've ensured support for all filter actions and have added corresponding tests for case-sensitive/insensitive, numeric, boolean, and date filters. All tests are passing.

However, I encountered an issue with the SVG icon toggle for case sensitivity. Adding it caused test failures, so I opted to proceed with the switch toggle for now. I would appreciate your thoughts on resolving the SVG icon issue.

`on adding this code:`
```tsx
<Grid size={1}>
    <Tooltip title="Toggle Case Sensitivity">
        <IconButton onClick={toggleMatchCase} className="case-sensitive-toggle">
            <SvgIcon>
                <path d="M20 14c0-1.5-.5-2-2-2h-2v-1c0-1 0-1-2-1v9h4c1.5 0 2-.53 2-2zm-8-2c0-1.5-.53-2-2-2H6c-1.5 0-2 .5-2 2v7h2v-3h4v3h2zm-2-5h4V5h-4zm12 2v11c0 1.11-.89 2-2 2H4a2 2 0 0 1-2-2V9c0-1.11.89-2 2-2h4V5l2-2h4l2 2v2h4a2 2 0 0 1 2 2m-6 8h2v-3h-2zM6 12h4v2H6z" />
            </SvgIcon>
        </IconButton>
    </Tooltip>
</Grid>
```

Tests are failing.

**Comment by Rishi-0007** [comment]
@jrobinAV Please add the labels for hacktoberfest, as a Hacktoberfest participant this would be highly useful for me. Thank you for your time.

**Comment by FredLL-Avaiga** [comment]
Please commit the SVG icon so that we can check it ...

**Comment by Rishi-0007** [comment]
Hi @FredLL-Avaiga,

I've added the requested code and also created a PR (#1975) for another issue. Could you kindly review it at your earliest convenience? Hacktoberfest is approaching its deadline, and I would greatly appreciate it if this PR could be merged soon. I am more than happy to make any further changes required to meet the standards.

Thank you very much for your time and support!

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 15]
```diff
@@ -126,44 +134,56 @@ const FilterRow = (props: FilterRowProps) => {
...
+    // Function to handle case-sensitivity toggle
+    const toggleMatchCase = useCallback(() => {
+        setMatchCase((prev) => !prev);
+    }, []);
+
     const onColSelect = useCallback(
         (e: SelectChangeEvent<string>) => {
             setColId(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, e.target.value, action, val));
+            setEnableCheck(!!getFilterDesc(columns, e.target.value, action, val, matchCase));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: don't pass it here

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 16]
```diff
@@ -126,44 +134,56 @@ const FilterRow = (props: FilterRowProps) => {
...
         },
         [columns, action, val]
     );
+
     const onActSelect = useCallback(
         (e: SelectChangeEvent<string>) => {
             setAction(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, colId, e.target.value, val));
+            setEnableCheck(!!getFilterDesc(columns, colId, e.target.value, val, matchCase));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: idem

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 17]
```diff
@@ -126,44 +134,56 @@ const FilterRow = (props: FilterRowProps) => {
...
         },
         [columns, colId, val]
     );
+
     const onValueChange = useCallback(
         (e: ChangeEvent<HTMLInputElement>) => {
             setVal(e.target.value);
-            setEnableCheck(!!getFilterDesc(columns, colId, action, e.target.value));
+            setEnableCheck(!!getFilterDesc(columns, colId, action, e.target.value, false));
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: don't pass it

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 18]
```diff
@@ -283,6 +304,17 @@ const FilterRow = (props: FilterRowProps) => {
                     />
                 )}
             </Grid>
+
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx:322]: I would add the icon only as an end Adornment to the AutoComplete and TextField within a MUi switch component

```jsx
<TextField ...
                        slotProps={{
                            input: {
                                endAdornment: (
                                    <Switch
                                        onChange={toggleMatchCase}
                                        checked={matchCase}
                                        className="case-sensitive-toggle"
                                        size="small"
                                        checkedIcon={<MatchCase />}
                                        icon={<MatchCase color="disabled" />}
                                    />
                                ),
                            },

```

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx:322]: with Matchcase in a frontend/taipy-gui/src/components/icons/MatchCase.tsx file (defined as 
```jsx
import { SvgIcon, SvgIconProps } from "@mui/material";

export const MatchCase = (props: SvgIconProps) => (
    <SvgIcon {...props} viewBox="0 0 16 16">
        <g stroke="currentColor">
        <path d="M20 14c0-1.5-.5-2-2-2h-2v-1c0-1 0-1-2-1v9h4c1.5 0 2-.53 2-2zm-8-2c0-1.5-.53-2-2-2H6c-1.5 0-2 .5-2 2v7h2v-3h4v3h2zm-2-5h4V5h-4zm12 2v11c0 1.11-.89 2-2 2H4a2 2 0 0 1-2-2V9c0-1.11.89-2 2-2h4V5l2-2h4l2 2v2h4a2 2 0 0 1 2 2m-6 8h2v-3h-2zM6 12h4v2H6z" />
        </g>
    </SvgIcon>
);

```

File: taipy/gui/data/pandas_data_accessor.py [hunk 19]
```diff
@@ -267,12 +267,21 @@ def __get_data(  # noqa: C901
                 col = fd.get("col")
                 val = fd.get("value")
                 action = fd.get("action")
+                match_case = fd.get("matchCase", False)
+
                 if isinstance(val, str):
-                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py:271]: Why did you remove these lines about date col ?

File: taipy/gui/data/test_pandas_accessor.py
Comment (FredLL-Avaiga) [inline taipy/gui/data/test_pandas_accessor.py:1]: this should go into tests/gui/data/test_pandas_data_accessor.py

File: taipy/gui/data/test_pandas_accessor.py [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+def sample_df():
+    data = {
+        "StringCol": ["Apple", "Banana", "Cherry", "apple"],
+        "NumberCol": [10, 20, 30, 40],
+        "BoolCol": [True, False, True, False],
+        "DateCol": pd.to_datetime(["2020-01-01", "2021-06-15", "2022-08-22", "2023-03-05"])
+    }
+    return pd.DataFrame(data)
+
+def test_contains_case_sensitive(pandas_accessor, sample_df):
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/test_pandas_accessor.py]: What about strings conparison  (not contains) ?

File: taipy/gui/data/pandas_data_accessor.py [hunk 21]
```diff
@@ -267,12 +267,21 @@ def __get_data(  # noqa: C901
...
-                        val = datetime.fromisoformat(val[:-1])
+                    if action == "contains":
+                        if not match_case:
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: you're still not dealing with string comparison (not contains) and matchcase

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

**Comment by Rishi-0007** [comment]
Hey @FredLL-Avaiga ,

I have completed the changes you requested, which include the following:

1. **Frontend:** Implemented the SVG icon as per your specifications. Additionally, I updated the frontend tests as per the new change. Also instead of using `className` I've used `data-testid`, as the tests were not recognizing `className` effectively. With this change, all frontend tests are passing successfully.
  
2. **Backend:** I updated the filtering logic in the backend to support proper case-insensitive string operations and date filtering. I also added corresponding tests to cover these functionalities. All backend tests are passing as well.

Let me know if anything else is needed!

**Comment by Rishi-0007** [comment]
Hi @jrobinAV @FlorianJacta,

As a Hacktoberfest participant, I would appreciate it if you could add the `hacktoberfest` and `hacktoberfest-200` labels to this PR as it addresses issue #426. This would help me showcase my contributions during the event. 

Thank you for your time and consideration.

File: taipy/gui/data/pandas_data_accessor.py [hunk 22]
```diff
@@ -269,25 +269,46 @@ def __get_data(  # noqa: C901
                 action = fd.get("action")
                 match_case = fd.get("matchCase", False)
 
-                if isinstance(val, str):
+                # Handle date columns separately from query
+                if self.__is_date_column(t.cast(pd.DataFrame, df), col):
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: we should test is date column only if val is a string

File: taipy/gui/data/pandas_data_accessor.py [hunk 23]
```diff
@@ -269,25 +269,46 @@ def __get_data(  # noqa: C901
...
+                        df = df[df[col] == val]
+                    is_copied = True
+                    continue  # Skip query for dates since it's already applied
+
+                elif isinstance(val, str):
+                    # Handle string operations
                     if action == "contains":
                         if not match_case:
-                            val = val.lower()  # Convert value to lowercase for case-insensitive filtering
+                            col = f"`{col}`.str.lower()"
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: please re-use code and do not duplicate

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
try to not duplicate code in pandas_data_accessor

**Comment by FredLL-Avaiga** [comment]
please update your branch

**Comment by FredLL-Avaiga** [comment]
Branch updated

**Comment by Rishi-0007** [comment]
Hi @FredLL-Avaiga , 
I've updated the `pandas_data_accessor.py` as per your feedback. 
Please check and review. Let me know if any further changes are needed.
Thank you for your time.

File: taipy/gui/data/pandas_data_accessor.py [hunk 24]
```diff
@@ -267,18 +267,47 @@ def __get_data(  # noqa: C901
                 col = fd.get("col")
                 val = fd.get("value")
                 action = fd.get("action")
-                if isinstance(val, str):
-                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
-                        val = datetime.fromisoformat(val[:-1])
+                match_case = fd.get("matchCase", False)
+
+                # Only test if the column is a date column if val is a string
+                if isinstance(val, str) and self.__is_date_column(t.cast(pd.DataFrame, df), col):
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: why do you change the previous code that was working ?
We want to build a query and apply it once.
the previous code
```py
                if isinstance(val, str):
                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
                        val = datetime.fromisoformat(val[:-1])
                    vars.append(val)
                val = f"@vars[{len(vars) - 1}]" if isinstance(val, (str, datetime)) else val
                right = f".str.contains({val})" if action == "contains" else f" {action} {val}"

```

The new code should be 
```py
                match_case = fd.get("matchCase", False) is not False
                right = None
                if isinstance(val, str):
                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
                        val = datetime.fromisoformat(val[:-1])
                    elif not match_case:
                        if action != "contains":
                            col = f"`{col}`.str.lower()"
                        val = val.lower()
                    vars.append(val)
                    val = f"@vars[{len(vars) - 1}]"
                    if action == "contains":
                        right = f".str.contains({val}{'' if match_case else ', case=False'})"
                if right is None:
                    right = f" {action} {val}"
```

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Can you update your branch and fix the tests ?

File: taipy/gui/data/pandas_data_accessor.py (hunk 24 above)
Comment (Rishi-0007) [reply taipy/gui/data/pandas_data_accessor.py]: Apologies for such mistakes, I have made the requested changes.
Please check and review it. Let me if further changes are needed.

Comment (Rishi-0007) [reply taipy/gui/data/pandas_data_accessor.py]: Also all the existing tests in `test_pandas_data_accessor.py` are passing after the changes made.

**Comment by Rishi-0007** [comment]
Hi @jrobinAV @FlorianJacta ,

As a Hacktoberfest participant, I would appreciate it if you could add the `hacktoberfest` and `hacktoberfest-200` labels to this PR as it addresses issue #426. This would help me showcase my contributions during the event. 

Thank you for your time and consideration.

**Review by FredLL-Avaiga — APPROVED** [review]
Looks good
I'll wait for the tests

File: frontend/taipy-gui/src/components/icons/MatchCase.tsx [hunk 25]
```diff
@@ -0,0 +1,9 @@
+import { SvgIcon, SvgIconProps } from "@mui/material";
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/icons/MatchCase.tsx]: `import SvgIcon, { SvgIconProps } from "@mui/material/SvgIcon";` for better tree shaking

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 26]
```diff
@@ -33,6 +33,9 @@ import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFnsV3";
 import { ColumnDesc, defaultDateFormat, getSortByIndex, iconInRowSx, FilterDesc } from "./tableUtils";
 import { getDateTime, getTypeFromDf } from "../../utils";
 import { getSuffixedClassNames } from "./utils";
+import SvgIcon from "@mui/icons-material/Check";
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: wrong and not needed ?

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 27]
```diff
@@ -33,6 +33,9 @@ import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFnsV3";
...
+import { FormControlLabel, Switch } from "@mui/material";
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: use current coding rules : specific import for better tree shaking
```
import FormControlLabel from "@mui/material/FormControlLabel";
import Switch from "@mui/material/Switch";

```

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: and separate node_modules imports from local imports (ordered)

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 28]
```diff
@@ -280,9 +302,24 @@ const FilterRow = (props: FilterRowProps) => {
...
+                        slotProps={{
+                            input: {
+                                endAdornment: (
+                                    <Switch
+                                        onChange={toggleMatchCase}
+                                        checked={matchCase}
+                                        size="small"
+                                        checkedIcon={<MatchCase />}
+                                        icon={<MatchCase color="disabled" />}
+                                        data-testid="CaseSensitiveToggle"
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: I don't like testid ...

File: frontend/taipy-gui/src/components/icons/MatchCase.tsx (hunk 25 above)
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/icons/MatchCase.tsx]: missing copyright header

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Almost there

**Comment by FredLL-Avaiga** [comment]
please fix linter issues

File: frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx [hunk 29]
```diff
@@ -48,7 +48,8 @@ beforeEach(() => {
 });
 
 afterEach(() => {
-    // @ts-ignore
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx:51]: ❓

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 30]
```diff
@@ -92,7 +94,14 @@ const getActionsByType = (colType?: string) =>
     (colType && colType in actionsByType && actionsByType[colType]) ||
     (colType === "any" ? { ...actionsByType.string, ...actionsByType.number } : actionsByType.string);
 
-const getFilterDesc = (columns: Record<string, ColumnDesc>, colId?: string, act?: string, val?: string) => {
+const getFilterDesc = (
+    columns: Record<string, ColumnDesc>,
+    colId?: string,
+    act?: string,
+    val?: string,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: remove this 
If we don't use it, it *is* an error ?

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: matchCase should be part of the returned object

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
we need to send the matchCase value to the backend !

File: frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx (hunk 29 above)
Comment (Rishi-0007) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx:51]: I was trying to fix linter issue, but I guess this was not the problem.

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx:51]: linter can be usefull ...

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx (hunk 30 above)
Comment (Rishi-0007) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: > remove this If we don't use it, it _is_ an error ?

ESlint throws an error of unused variable.

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: well it means that you should use it !

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: to be more obvious:
```
             type: colType,
            } as FilterDesc;
```

should be
```

             type: colType,
             matchCase: !!matchCase,
            } as FilterDesc;
```

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx:1]: can you fix the linter issues ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx
Comment (Rishi-0007) [reply frontend/taipy-gui/src/components/Taipy/TableFilter.spec.tsx:1]: > can you fix the linter issues ?

I am working on it.

**Review by FredLL-Avaiga — APPROVED** [review]
Looks good tx 
@jrobinAV I think we can merge this

**Comment by Rishi-0007** [comment]
Hi @FredLL-Avaiga @jrobinAV ,

Thank you so much for merging my first PR! in any open source organization. I really appreciate your feedback and support throughout the process. I’m excited to keep contributing and learning.

Best,
Rishi Nayak

**Comment by RymMichaut** [comment]
Congratulations, @Rishi-0007, for this merged Hacktoberfest PR as well.
Same, contact me please at rym.michaut@taipy.io
